clients/rust/README.md

```diff
@@ -0,0 +1,276 @@
+ ## Getting started
+
+From your project folder:
+ ```bash
+ cargo add mpl-token-metadata
+ ```
+
+ ## Structure
+
+ The client SDK is divided into several modules:
+
+ * `accounts`: structs representing the accounts of the program
+ * `errors`: enums representing the program errors
+ * `instructions`: structs to facilitate the creation of instructions, instruction arguments and CPI instructions
+ * `types`: structs representing types used by the program
+
+## Instruction Builders
+
+One of the main features of the client SDK is to facilitate the creation of instructions. There are two "types" of instruction builders automatically generated – both support passing accounts by name and optional positional 
+
+### _Client_ instruction builders
+
+This are intended to be used by off-chain client code. Each instruction is represented by a corresponding struct – e.g., `CreateV1`:
+```rust
+pub struct CreateV1 {
+    /// Unallocated metadata account with address as pda of ['metadata', program id, mint id]
+    pub metadata: solana_program::pubkey::Pubkey,
+    /// Unallocated edition account with address as pda of ['metadata', program id, mint, 'edition']
+    pub master_edition: Option<solana_program::pubkey::Pubkey>,
+    /// Mint of token asset
+    pub mint: (solana_program::pubkey::Pubkey, bool),
+    /// Mint authority
+    pub authority: solana_program::pubkey::Pubkey,
+    /// Payer
+    pub payer: solana_program::pubkey::Pubkey,
+    /// Update authority for the metadata account
+    pub update_authority: (solana_program::pubkey::Pubkey, bool),
+    /// System program
+    pub system_program: solana_program::pubkey::Pubkey,
+    /// Instructions sysvar account
+    pub sysvar_instructions: solana_program::pubkey::Pubkey,
+    /// SPL Token program
+    pub spl_token_program: solana_program::pubkey::Pubkey,
+}
+```
+
+After filling in the instruction account fields, you can use the `instruction(...)` method to generate the corresponding `solana_program::instruction::Instruction`:
+```rust
+// instruction args
+let args = CreateV1InstructionArgs {
+    name: String::from("pNFT"),
+    symbol: String::from(""),
+    uri: String::from("http://my.pnft"),
+    seller_fee_basis_points: 500,
+    primary_sale_happened: false,
+    is_mutable: true,
+    token_standard: TokenStandard::ProgrammableNonFungible,
+    collection: None,
+    uses: None,
+    collection_details: None,
+    creators: None,
+    rule_set: None,
+    decimals: Some(0),
+    print_supply: Some(PrintSupply::Zero),
+};
+
+// instruction accounts
+let create_ix = CreateV1 {
+    metadata,
+    master_edition: Some(master_edition),
+    mint: (mint_pubkey, true),
+    authority: payer_pubkey,
+    payer: payer_pubkey,
+    update_authority: (payer_pubkey, true),
+    system_program: system_program::ID,
+    sysvar_instructions: solana_program::sysvar::instructions::ID,
+    spl_token_program: spl_token::ID,
+};
+
+// creates the instruction
+let create_ix = create_instruction(args);
```

Comment:
Shouldn't this be something like `let create_ix = create_ix.instruction(args)`?

Reply:
Yes, fixed the typo – thanks!